src/test/java/com/learning/pages/google/FirstSeleniumTest.java

```diff
@@ -29,8 +42,8 @@ void shouldGoToJunit5WebPage() {
   @Test
   void shouldFindElementOnGoogleWebSite() {
```

Comment:
Try implementing stuff, so the test looks like this:

```java
void shouldFindElementOnGoogleWebSite() {
    googleSearchPage.open().search("Getting grip on it!");
}
```
(GoogleSearchPage would have to be injected by Guice)

This way:
- googleSearchPage is responsible for knowing it's own URL
- googleSearchPage holds logic responsible for waiting for page to load
- googleSearchPage holds logic about search business process
- test is more readable
- test contains only business steps, high level ones, not all interactions with browser
- no selenium in tests (no WebElements, no driver)
- close to none setup
- tearDown could also be removed if cleanup was moved to `Browser` class and `PreDestroy` method.

Reply:
I have got close to requirements you provided but I think I need to see your `Browser` class to understand it. 
But let's wait with that until I finish the rest.